### PR TITLE
FormBrowse Commands in toolbar menu raised exceptions for artificial …

### PR DESCRIPTION
**Pull request by gerhardol**
…commits and no revisions at all

Related to #4031 and somewhat to #4098

The irrelevant commands are disabled
There are some existing checks for bareRepositories in InternalInitialize() (where some init code is running...) that also could be removed after this (some were missing from that menu). checkoutBranchToolStripMenuItem is a little special though.

Screenshots before and after (if PR changes UI):
- 
image

- 
image

How did I test this code:
 - Empty, bare and artificial commits. (There may be some corner case to not select a rev also in other situation that I do not know that applies too)
 - Select Commands menu

Has been tested on (remove any that don't apply):
 - Windows 10

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -2017,6 +2012,23 @@ protected override void OnClosed(EventArgs e)
             base.OnClosed(e);
         }
 
+        private void CommandsToolStripMenuItem_DropDownOpening(object sender, System.EventArgs e)
+        {
+            //Most options do not make sense for artificial commits or no revision selected at all
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
+            bool enabled = selectedRevisions.Count == 1 && !selectedRevisions.Single().IsArtificial();
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormBrowse.cs]: `selectedRevisions[0]` is more efficient than `selectedRevisions.Single()`

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 2]
```diff
@@ -2017,6 +2012,23 @@ protected override void OnClosed(EventArgs e)
             base.OnClosed(e);
         }
 
+        private void CommandsToolStripMenuItem_DropDownOpening(object sender, System.EventArgs e)
+        {
+            //Most options do not make sense for artificial commits or no revision selected at all
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
+            bool enabled = selectedRevisions.Count == 1 && !selectedRevisions[0].IsArtificial();
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/FormBrowse.cs]: Bug here! You forgot to check that is not bare repository

Comment (gerhardol) [reply GitUI/CommandsDialogs/FormBrowse.cs]: Noted. I have planned another review after #4157 and at least one PR building on that is merged.
There are still some situations that should be disabled.

Are there any users with bare repositries? Why use a GUI at all?

Comment (KindDragon) [reply GitUI/CommandsDialogs/FormBrowse.cs]: I rarely, just preferring use GUI for everything.
